lebedigital/openbis/interbis.py

```diff
@@ -659,3 +682,70 @@ def combine_urls(base_url, relative_url):
 
         response = requests.post(url=combine_urls(self.url, self.as_v3), json=request, verify=self.verify_certificates).json()
         return response
+
+    def _get_datatype_conversion(self, property_name: str, property_datatype: str):
+        """
+        Converts the openbis datatypes into python datatypes if possible, else a custom datatype
+        Use with the `generate_typechecker` method
+        """
+        # if the prop is in the dict then it is not a CONTROLLED_VOCABULARY
+        if property_datatype in CONVERSION_DICT:
+            return CONVERSION_DICT[property_datatype]
+
+        vocabulary_name = self.get_property_type(property_name).vocabulary
```

Comment:
what exactly is happening here, could you document that answer as well?

Reply:
I added a comment in that spot. What is happening is that we can generally just simply cast props to their python types. For example the types VARCHAR, REAL or INTEGER to string, int and float respectively so a simple dict lookup to a static dict is enough for pydantic to know what it needs to get. Controlled Vocabularies are the only data type that is not in that dict because it is the only data type which needs special treatment - constructing an Enum with their accepted values which we then run against our generated/read data we want to upload to openbis.